lightning/src/util/persist.rs

```diff
@@ -243,10 +250,886 @@ where
+			CHANNEL_MONITOR_PERSISTENCE_NAMESPACE,
+			CHANNEL_MONITOR_PERSISTENCE_SUB_NAMESPACE,
+		)?;
+		for maybe_monitor_name in monitor_list.into_iter().map(MonitorName::new) {
+			let monitor_name = maybe_monitor_name?;
+			let (_, monitor) = self.read_monitor(&monitor_name)?;
+			let end = monitor.get_latest_update_id();
+			let start = end.saturating_sub(search_depth);
+			self.lazy_delete_updates(&monitor_name, start, end)?;
+		}
+		Ok(())
+	}
+}
+
+impl<
+		ChannelSigner: WriteableEcdsaChannelSigner,
+		K: Deref,
+		L: Deref,
+		ES: Deref + Clone,
+		SP: Deref + Clone,
+	> Persist<ChannelSigner> for MonitorUpdatingPersister<K, L, ES, SP>
+where
+	K::Target: KVStore,
+	L::Target: Logger,
+	ES::Target: EntropySource + Sized,
+	SP::Target: SignerProvider + Sized,
+{
+	/// Persists a new channel. This means writing the entire monitor to the
+	/// provided [`KVStore`].
+	fn persist_new_channel(
+		&self,
+		funding_txo: OutPoint,
+		monitor: &ChannelMonitor<ChannelSigner>,
+		_monitor_update_call_id: MonitorUpdateId,
+	) -> chain::ChannelMonitorUpdateStatus {
+		// Determine the proper key for this monitor
+		let monitor_name = match MonitorName::try_from(funding_txo) {
+			Ok(n) => n,
+			Err(_) => return chain::ChannelMonitorUpdateStatus::InProgress,
+		};
+		let key = monitor_name.as_str();
+		// Check that this key isn't already storing a monitor with a higher update_id (collision)
+		let maybe_old_monitor = self.read_monitor(&monitor_name);
+		match maybe_old_monitor {
+			Ok((_, ref old_monitor)) => {
+				if old_monitor.get_latest_update_id() > monitor.get_latest_update_id() {
+					return chain::ChannelMonitorUpdateStatus::PermanentFailure
+				}
+			},
+			Err(ref e) if e.kind() == io::ErrorKind::NotFound => {},
+			_ => return chain::ChannelMonitorUpdateStatus::InProgress
+		}
+		// Serialize and write the new monitor
+		let mut monitor_bytes = Vec::with_capacity(
+			MONITOR_UPDATING_PERSISTER_PREPEND_SENTINEL.len() + monitor.serialized_length(),
+		);
+		monitor_bytes.extend_from_slice(MONITOR_UPDATING_PERSISTER_PREPEND_SENTINEL);
+		monitor.write(&mut monitor_bytes).unwrap();
+		match self.kv_store.write(
+			CHANNEL_MONITOR_PERSISTENCE_NAMESPACE,
+			CHANNEL_MONITOR_PERSISTENCE_SUB_NAMESPACE,
+			key,
+			&monitor_bytes,
+		) {
+			Ok(_) => {
+				// Assess cleanup
+				if let Ok((_, old_monitor)) = maybe_old_monitor {
+					// Typically, we'll clean up between the last two known full monitors
+					let start = old_monitor.get_latest_update_id();
+					let mut end = monitor.get_latest_update_id();
+					if monitor.get_latest_update_id() == CLOSED_CHANNEL_UPDATE_ID {
+						// We don't want to clean the rest of u64, so just do possible pending updates
+						end = start.saturating_add(self.maximum_pending_updates);
+					}
+					if let Err(e) =
+						self.lazy_delete_updates(&MonitorName::from(funding_txo), start, end)
+					{
+						log_error!(
+							self.logger,
+							"error cleaning up channel monitor updates {}",
```

Comment:
Could you add a `:` before the error in this function and the next?

Reply:
thanks, done